### DISCUSSION TITLE
Matrix room

### DISCUSSION THREAD

**Discussion by Xiretza** [blackmagic-debug/blackmagic#1186] [Ideas]
Hey, I'm glad to see this project being officially revived from maintenance stasis! I'd love to contribute or at least have a chat, but unfortunately the only official communication channel (besides github issues/discussions) seems to be discord. Discord has many issues related to privacy, security and simply being a centralized and proprietary service; using it exclusively for a FOSS project does not seem appropriate to me.

Fortunately, there are viable alternatives: [Matrix](https://matrix.org/) is a federated, open protocol that has already been adopted as a communication medium by many FOSS communities. It even offers the possibility of bridges, such as the one [hosted by t2bot.io](https://t2bot.io/discord/), which allow other protocols to be seamlessly integrated into a room.

My suggestion: Create a matrix room, then bridge it to the existing discord room using the t2bot bridge. This means no change to discord users, but allows new users to enjoy the chat without ever having to directly interact with discord. From personal experience, this method has worked great for the [OpenRTX](https://openrtx.org/#/get_in_touch) project so far.

**Comment by esden** [comment]
Hi @Xiretza,

I hear your argument. But discord works well for our purposes. As any open source project it is important to realize what we spend our time on and not to spread out our efforts and communications too thin. Bridges bring their own set of issues with them. We already maintain bridges for some of the discord channels to IRC. As those channels already existed. You might be surprised to hear that it takes a lot of effort to keep them running. Adding Matrix also to the mix will make things exponentially more complicated.

We are an open source project working on JTAG/SWD debugging. We can't solve all the open-source problems. So as a maintainer I will not take on the "problem" of chats. Discord works very well. Your privacy criticism is also not really valid, these are open chats, publicly available. If you want to keep privacy you can use tunnels and nicknames. Discord in this case is just another chat platform. IRC or Matrix would not be more or less "private" in this case. If Discord stops serving our needs or becomes hard to use for the community members we will consider moving to some other service. As we already did with gitter, that fell into disrepair and became really hard to use.

I don't blame you if you personally don't want to use Discord, but for the rest of us it does the job and keeps things convenient. Let's try to concentrate our efforts on moving BMD forward and not get too distracted by everything else that is not core to the project misson.

Cheers,
Piotr
